http-response/background.js

```diff
@@ -1,22 +1,154 @@
 function listener(details) {
+  // The received data is a stream of bytes. In order to do text-based
+  // modifications, it is necessary to decode the bytes into a string
+  // using the proper character encoding, do any modifications, then
+  // encode back into a stream of bytes.
+  // Historically, detecting character encoding has been a tricky task
+  // taken on by the browser. Here, a simplified approach is taken
+  // and the complexity is hidden in a helper method.
+  let decoder, encoder;
+  [decoder, encoder] = detectCharsetAndSetupDecoderEncoder(details);
   let filter = browser.webRequest.filterResponseData(details.requestId);
-  let decoder = new TextDecoder("utf-8");
-  let encoder = new TextEncoder();
+  let fullStr = '';
+  
+  filter.ondata = e => {
+    // Note that the event's data may break in the middle of an encoded
+    // character - the stream parameter is critical for success as this
+    // method gets called multiple times.
+    let str = decoder.decode(e.data, {stream: true});
+    fullStr += str;
+  }
+  
+  filter.onstop = async e => {
+    fullStr += decoder.decode(); //Flush the buffer
+    // Just change any instance of Example or Test in the HTTP response
+    // to WebExtension Example or WebExtension Test.
+    let mutatedStr = fullStr.replace(/Example/g, 'WebExtension Example');
+    mutatedStr = mutatedStr.replace(/Test/g, 'WebExtension Test');

```

Comment:
Remove the second one. It is not necessary for the example.

Reply:
It is necessary for the example. I added it because the W3C test suite pages do not have the word "Example" but do have "Test" so it shows that the plugin is mutating the pages when viewing the test suite - I think that's a critical aspect to preserve. However, in the name of simplicity, I could remove the "Example"/example.com from the addon; I was trying to keep a nod to the original behavior but perhaps it just feels kludgy now.

Reply:
If you'd like to replace both, you could do `/Example|Test/` to demonstrate how one can replace multiple words (and if you wish to prepend "WebExtension", use `'WebExtension $&'`. If you don't know about `$&`, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter

Reply:
Thanks - I think your first comment made me realize that I had a conflict of "make everyone happy" or "have a better focused example" and so I'm choosing the latter and keeping it simple. As venerable as example.com is, it gets enough press elsewhere.